api/include/opentelemetry/nostd/span.h

```diff
@@ -5,7 +5,27 @@
 
 // Try to use either `std::span` or `gsl::span`
 #ifdef HAVE_CPP_STDLIB
-#  include "opentelemetry/std/span.h"
+#  include <array>
+#  include <cstddef>
+#  include <iterator>
+#  include <type_traits>
+
+/**
+ * @brief Clang 14.0.0 with libc++ do not support implicitly construct a span
+ * for a range. We just use our fallback version.
+ *
+ */
+#  if defined(_LIBCPP_VERSION)
+#    if _LIBCPP_VERSION <= 14000
+#      define OPENTELEMETRY_OPTION_USE_STD_SPAN 0
+#    endif
+#  endif
+#  ifndef OPENTELEMETRY_OPTION_USE_STD_SPAN
+#    define OPENTELEMETRY_OPTION_USE_STD_SPAN 1
+#  endif
+#  if OPENTELEMETRY_OPTION_USE_STD_SPAN
+#    include "opentelemetry/std/span.h"
+#  endif
```

Comment:
nit  - do you think we should document the fallback scenario, maybe here - https://github.com/open-telemetry/opentelemetry-cpp/blob/main/docs/dependencies.md#external-dependencies where we talk about HAVE_CPP_STDLIB  and WITH_STL ?

Reply:
> . I just added some notes for this case into https://github.com/open-telemetry/opentelemetry-cpp/blob/main/docs/dependencies.md . 

Thanks, I don't see the markdown updated as part of this PR, just in case missed committing it :)

>Do you think if we should add a standalone config file to indicate whether some components could be alias for STL version? 

You mean the config file will have all platform specific code which set the required macros ( e.g. OPENTELEMETRY_OPTION_USE_STD_SPAN ), and it would be included within all nostd:: implementation ? We can plan for any such implementation separate to this PR if that simplifies the code.

Reply:
Sorry, I forget to push commits before.